scripts/install-nbd-kmod.sh

```diff
@@ -13,37 +13,43 @@ KURL="http://vault.centos.org/7.6.1810/updates/Source/SPackages/$KRPM"
 
 GENERICBUILDDIR=$(pwd)/build
 NBDBUILDDIR=$GENERICBUILDDIR/nbdbuild
+KBUILDDIR=$NBDBUILDDIR/BUILD/kernel-$KSRC/linux-$KSRC.x86_64
+OUTPUTFILE=$GENERICBUILDDIR/nbd.ko
 
 mkdir -p $GENERICBUILDDIR
 cd $GENERICBUILDDIR
 echo $(pwd)
 mkdir -p $NBDBUILDDIR/{BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
 
+rm -rf "$NBDBUILDDIR"
+rm -rf "$KRPM"* $OUTPUTFILE
 wget $KURL
 
 # unpack
+cd $GENERICBUILDDIR
 rpm --define="_topdir $NBDBUILDDIR" -ivh $KRPM
 
 cd $NBDBUILDDIR/SPECS
 rpmbuild --define="_topdir $NBDBUILDDIR" -bp --target=$(uname -m) kernel.spec
 
-KBUILDDIR=$NBDBUILDDIR/BUILD/kernel-$KSRC/linux-$KSRC.x86_64/
-cd $KBUILDDIR
 
 # this file is not kept up to date and does not compile, need to patch it
+cd $KBUILDDIR
 sed -i 's/REQ_TYPE_SPECIAL/REQ_TYPE_DRV_PRIV/g' $KBUILDDIR/drivers/block/nbd.c
 
-# now build
-make -j32 prepare
-make -j32 modules_prepare
+# now clean/setup .config
+cd $KBUILDDIR
+make -j32 mrproper
+make -j32 oldconfig
```

Comment:
Just to be clear, this is a fairly significant functional change to the flow:
* `mrproper` removes the kernel config initialized by the RPM `%prep` stage.  This is the config that ships with the source RPM, which I expect to match the binary package deployed in the AMI.
* `oldconfig` replaces it with a config taken from the kernel currently running on the build host, usually `/boot/config-$(uname -r)`.

On a fresh manager instance, the two configs should be identical (but need to confirm that).  However, they can diverge if the user ran `yum upgrade` on the manager before the build.

Reply:
Also, `oldconfig` is potentially interactive.  If an option is undefined, it prompts the user to select its value.  Instead use `olddefconfig` to automatically choose the default.